actors/init/src/lib.rs

```diff
@@ -129,11 +136,25 @@ impl Actor {
 
         // Allocate an ID for this actor.
         // Store mapping of actor addresses to the actor ID.
-        let id_address: ActorID = rt.transaction(|s: &mut State, rt| {
+        let (id_address, existing): (ActorID, bool) = rt.transaction(|s: &mut State, rt| {
             s.map_addresses_to_id(rt.store(), &robust_address, Some(&delegated_address))
                 .context("failed to map addresses to ID")
         })?;
 
+        // If the f4 address was already assigned, make sure we're deploying over an embryo and not
+        // some other existing actor (and make sure the target actor wasn't deleted either).
+        if existing {
+            let code_cid = rt
+                .get_actor_code_cid(&id_address)
+                .context_code(ActorError::USR_FORBIDDEN, "cannot redeploy a deleted actor")?;
+            let embryo_cid = rt.get_code_cid_for_type(Type::Embryo);
+            if code_cid != embryo_cid {
+                return Err(ActorError::forbidden(format!(
+                    "cannot replace an existing non-embryo actor with code: {code_cid}"
+                )));
+            }
```

Comment:
Technically the FVM will enforce this second part. But we might as well double check.